packages/firebase_auth/firebase_auth_platform_interface/lib/src/method_channel_firebase_auth.dart

```diff
@@ -0,0 +1,459 @@
+  Future<PlatformIdTokenResult> getIdToken(String app, bool refresh) async {
+    final Map<String, dynamic> data = await channel
+        .invokeMapMethod<String, dynamic>('getIdToken', <String, dynamic>{
+      'refresh': refresh,
+      'app': app,
+    });
+
+    return _decodeIdTokenResult(data);
+  }
+
+  @override
+  Future<PlatformAuthResult> reauthenticateWithCredential(
+    String app,
+    AuthCredential credential,
+  ) async {
+    final Map<String, dynamic> data =
+        await channel.invokeMapMethod<String, dynamic>(
+      'reauthenticateWithCredential',
+      <String, dynamic>{
+        'app': app,
+        'provider': credential.providerId,
+        'data': credential.asMap(),
+      },
+    );
+    return _decodeAuthResult(data);
+  }
+
+  @override
+  Future<PlatformAuthResult> linkWithCredential(
+    String app,
+    AuthCredential credential,
+  ) async {
+    final Map<String, dynamic> data =
+        await channel.invokeMapMethod<String, dynamic>(
+      'linkWithCredential',
+      <String, dynamic>{
+        'app': app,
+        'provider': credential.providerId,
+        'data': credential.asMap(),
+      },
+    );
+    final PlatformAuthResult result = _decodeAuthResult(data);
+    return result;
+  }
+
+  @override
+  Future<void> unlinkFromProvider(String app, String provider) {
+    return channel.invokeMethod<void>(
+      'unlinkFromProvider',
+      <String, String>{'provider': provider, 'app': app},
+    );
+  }
+
+  @override
+  Future<void> updateEmail(String app, String email) {
+    return channel.invokeMethod<void>(
+      'updateEmail',
+      <String, String>{'email': email, 'app': app},
+    );
+  }
+
+  @override
+  Future<void> updatePhoneNumber(
+    String app,
+    PhoneAuthCredential phoneAuthCredential,
+  ) {
+    return channel.invokeMethod<void>(
+      'updatePhoneNumberCredential',
+      <String, dynamic>{
+        'app': app,
+        'provider': phoneAuthCredential.providerId,
+        'data': phoneAuthCredential.asMap(),
+      },
+    );
+  }
+
+  @override
+  Future<void> updatePassword(String app, String password) {
+    return channel.invokeMethod<void>(
+      'updatePassword',
+      <String, String>{'password': password, 'app': app},
+    );
+  }
+
+  @override
+  Future<void> updateProfile(
+    String app, {
+    String displayName,
+    String photoUrl,
+  }) {
+    final Map<String, String> arguments = <String, String>{'app': app};
+    if (displayName != null) {
+      arguments['displayName'] = displayName;
+    }
+    if (photoUrl != null) {
+      arguments['photoUrl'] = photoUrl;
+    }
+    return channel.invokeMethod<void>(
+      'updateProfile',
+      arguments,
+    );
+  }
+
+  @override
+  Future<void> setLanguageCode(String app, String language) {
+    return channel.invokeMethod<void>('setLanguageCode', <String, String>{
+      'language': language,
+      'app': app,
+    });
+  }
+
+  @override
+  Stream<PlatformUser> onAuthStateChanged(String app) {
```

Comment:
I'd expect an `onSomething` method to receive a callback or similar. Maybe `getPlatformUserStream`?

Reply:
In fact this naming style is common in [dart:html](https://api.dartlang.org/stable/2.6.1/dart-html/Element/onChange.html), but I personally find it extremely confusing :/

Reply:
And it matches the name of the method in `package:firebase_auth`